### PR TITLE
Behavior cloning

### PR DESCRIPTION
**Pull request by hugheste**
Most up-to-date version of the code for the reinforcement learning training and evaluation. Changes will be walked through by Stephanie so the current state of the code is clear.

### CODE REVIEW

File: gym-kinova-gripper/gym_kinova_gripper/envs/kinova_gripper_env.py [hunk 1]
```diff
@@ -1192,43 +1216,106 @@ def select_orienation(self, random_shape, hand_orientation):
...
-        else:
-            # Rotated (45 deg) orientation
+        elif hand_orientation == 'rotated':
+            # Rotated (67 deg) orientation
             orientation = 'rotated'
 
         return orientation
 
+    def convert_local_obj_coord_to_global(self,local_obj_x,local_obj_y,local_obj_z=None):
+        """ Convert a specific local x,y,(z is optional) object coordinate to the global representation. This allows
```
Comment (hugheste) [inline gym-kinova-gripper/gym_kinova_gripper/envs/kinova_gripper_env.py:1257]: Remove convert local obj coord to global function so there is no confusion

File: gym-kinova-gripper/gym_kinova_gripper/envs/kinova_gripper_env.py [hunk 2]
```diff
@@ -1262,6 +1348,9 @@ def determine_obj_hand_coords(self, random_shape, mode, with_noise=True):
                 new_rotation=np.array([0,0,0]) # Top
             else:
                 new_rotation=np.array([1.2,0,0]) # Rotated
+            hand_x = new_rotation[0]
```
Comment (hugheste) [inline gym-kinova-gripper/gym_kinova_gripper/envs/kinova_gripper_env.py:1351]: Turn hand_x, hand_y, hand_z into an array

File: gym-kinova-gripper/check_coordinates_for_collision.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            state = env.reset(shape_keys=requested_shapes, hand_orientation=hand_orientation, with_grasp=with_grasp,
+                               env_name="env", mode=mode, orient_idx=orient_idx, with_noise=with_noise)
+
+            # Print which file you're running through
+            if orient_idx == 0:
+                coords_file = env.get_coords_filename()
+                print("Coords filename: ", coords_file)
+
+            # Get the current object coordinate and hand orientation pair
+            obj_coords = env.get_obj_coords()
```
Comment (hugheste) [inline gym-kinova-gripper/check_coordinates_for_collision.py:73]: Needs to be updated: previous object coordinates are not being updated. Need to update this in order to correctly check for collision.

**Review by hugheste — COMMENTED** [review]
Files have been reviewed by Nigel and Stephanie -- collision detection will be updated otherwise the overall logic changes are up to date with the current representation of our episode structure and training structure.

**Review by nigelswenson — APPROVED** [review]
Reviewed with Stephanie over zoom. She has added all our comments.

**Pull request merged**